### PR TITLE
perf(test): cut renderer suite cost without dropping coverage

### PR DESCRIPTION
**Pull request by junyudev**
## Summary

- close every row in the renderer CI audit: 35 slow files plus 2 exceptional single tests
- move pure settings, request-context, copy-payload, and interrupted-resume rules into production-owned modules while retaining their representative DOM wiring contracts
- make settled renderer behavior tests complete Motion timelines immediately without impersonating reduced-motion preferences; full-motion and reduced-motion contracts opt in explicitly
- isolate independently covered calendar/editor children behind typed semantic adapters and keep focused real-calendar interaction coverage
- remove four presentation-only assertions while preserving ownership, failure recovery, focus, selection, portal, pagination, streaming, and Browser/WebView lifecycle coverage
- fix the request-card focus race exposed by synchronous reduced-motion mounting and keep all renderer tests React `act`-clean
- keep explicit full CI runs below Linux `ARG_MAX` by omitting the affected-path payload that all-gates runs do not consume

## Audit outcome

Three restarted samples on the same final tree each passed 37/37 files and 944/944 tests:

| Metric | PR #71 audit baseline | Final median |
| --- | ---: | ---: |
| 35 audited slow files, aggregate `tests` | 305.97s | 51.32s |
| WorkbenchShell nine-file group, aggregate `tests` | 192.45s | 27.29s |
| P0 property/date seams | 6.17s / 5.83s / 4.73s | 0.29s / 0.18s / 0.12s |

All 37 rows have an explicit outcome: an owning production module, a deterministic heavy-child/transition/data seam with retained DOM contracts, or a measured P3 acceptance where further refactoring would reduce clarity without meaningful value.

## Validation

- rebased onto current `origin/main`; range-diff preserved the pre-existing commit series exactly
- `pnpm run typecheck`
- `pnpm run lint`
- local full renderer after the final review fixes: 318/318 files and 2,488/2,488 tests passed in 134.94s
- full-renderer warning scan: no React `act`, Motion, missing-key, or undefined-query warnings
- three final audit samples: 37/37 files and 944/944 tests per sample
- Page Create portalled-picker stabilization: 10 complete-file runs, 140/140 tests
- final-head required PR CI passed: https://github.com/junyudev/nodex/actions/runs/32458770929
- final-head explicit `full=true` matrix passed: https://github.com/junyudev/nodex/actions/runs/32458822617
- both CodeRabbit findings were fixed, replied to, confirmed, and resolved

## Notes

- `CHANGELOG.md` is unchanged because this is internal test/CI infrastructure; the only product correction preserves the already-intended request focus behavior.
- `cargo-nextest 0.9.143` is installed locally for Rust iteration.
- GitHub larger 8-vCPU runners are billed for this repository, so this work stays on the free standard runner.

### CODE REVIEW

File: src/renderer/components/workbench/workbench-testkit/workbench-shell-harness.tsx [hunk 1]
```diff
@@ -3604,6 +3598,10 @@ export function installTerminalEventApiMock(): TerminalEventListenerMap {
 }
 
 beforeEach(() => {
+  restoreDefaultReducedMotion?.();
+  // Shell behavior tests assert settled product state. Avoid keeping Motion's
+  // exit trees and timers alive unless a test explicitly owns that contract.
+  restoreDefaultReducedMotion = installReducedMotionMatchMediaForTest();
```
[the remark this answers is not in the document]
Comment (junyudev) [reply src/renderer/components/workbench/workbench-testkit/workbench-shell-harness.tsx]: Fixed in f91a620d9. The Workbench harness now uses a dedicated neutral instant-Motion helper: prefers-reduced-motion remains false while MotionGlobalConfig.skipAnimations is true, with initPrefersReducedMotion refreshed on install and restore. Explicit reduced-motion and full-motion helpers remain separate. A regression test asserts both boundary values. Validated with 43 focused tests, three 944-test audit samples, typecheck, lint, and the warning-clean 318-file renderer suite. — on behalf of Jun by gpt-5.6-sol

File: src/renderer/features/browser-sidebar/browser-tab-favicon.test.tsx [hunk 2]
```diff
@@ -67,38 +68,43 @@ describe("BrowserTabFavicon", () => {
...
+      expect(view.container.querySelector("img")?.getAttribute("src"))
+        .toBe("https://example.com/favicon.ico");
+      const clip = view.container.querySelector<HTMLElement>(
+        "[data-browser-tab-favicon-clip='true']",
+      );
+      if (!clip) throw new Error("Expected favicon clip frame");
+
+      fireEvent.transitionEnd(clip, { propertyName: "clip-path" });
+
+      expect(view.container.querySelector("img")).toBeNull();
```
[the remark this answers is not in the document]
Comment (junyudev) [reply src/renderer/features/browser-sidebar/browser-tab-favicon.test.tsx]: Fixed in f91a620d9. The favicon error/transitionEnd events and every identified request-card option/navigation click now run in awaited act blocks, while DOM and focus assertions remain outside after React settles. Validated with 43 focused tests and the warning-clean 318-file, 2,488-test renderer suite. — on behalf of Jun by gpt-5.6-sol
